heron/schedulers/src/java/com/twitter/heron/scheduler/kubernetes/KubernetesScheduler.java

```diff
@@ -417,11 +418,13 @@ protected ObjectNode getContainerSpec(ObjectMapper mapper,
   protected ArrayNode getPorts(ObjectMapper mapper) {
     ArrayNode ports = mapper.createArrayNode();
 
-    for (int i = 0; i < KubernetesConstants.PORT_NAMES.length; i++) {
+    for (Map.Entry<ExecutorPort, String> entry
```

Comment:
Converting `KubernetesConstants.EXECUTOR_PORTS` to a `<String, Integer>` map would reduce one unnecessary `Integer.parseInt` below.

Reply:
I am not sure this is a good idea.  For some of the schedulers, e.g. Marathon Scheduler, the ports are actually strings, e.g. $PORT1.  Thus, its probably simpler if we just keep everything as a String.